nav2_bt_navigator/src/bt_navigator.cpp

```diff
@@ -259,6 +274,17 @@ BtNavigator::navigateToPose()
       action_server_->publish_feedback(feedback_msg);
     };
 
+  auto bt_xml_filename = action_server_->get_current_goal()->behavior_tree;
+
+  // Empty id in request is default for backward compatibility
+  bt_xml_filename = bt_xml_filename == "" ? default_bt_xml_filename_ : bt_xml_filename;
+
+  if (!loadBehaviorTree(bt_xml_filename)) {
```

Comment:
Only question I have here is whether we should fail the navigation task when the BT isn't valid rather than using the old tree. 

if these trees are very, very different like one is navigate to a point, and another is a multi-part task, having it default to the last one seems like bad behavior. I'd think we'd want to immediately fail the task with `terminate_current`. What do you think?

Reply:
Yes, please fail fast, early and loud.

Reply:
Yes, @SteveMacenski. I have just fixed this. Now it fails and terminates action if BT is not valid.